src/nvim/getchar.c

```diff
@@ -509,7 +509,7 @@ void AppendToRedobuffLit(const char *str, int len)
       s--;
     }
     if (s > start) {
-      add_buff(&redobuff, start, (long)(s - start));
+      add_buff(&redobuff, start, (long)(s - start));  // NOLINT(google-readability-casting)
```

Comment:
It seems that this cast can indeed be removed, as `add_buff()` accepts a `ptrdiff_t`.